FirebaseMLModelDownloader/Tests/Unit/ModelDownloaderUnitTests.swift

```diff
@@ -148,3 +144,376 @@ final class ModelDownloaderUnitTests: XCTestCase {
+  let fakeAPIKey = "fakeAPIKey"
+  var fakeRemoteModelInfo: String {
+    """
+    {
+      "downloadUri":"\(fakeDownloadURL)",
+      "expireTime":"\(fakeExpiryTime)",
+      "sizeBytes":"\(fakeModelSize)"
+    }
+    """
+  }
+
+  let successAuthTokenProvider =
+    { (completion: @escaping (Result<String, DownloadError>) -> Void) in
+      completion(.success("fakeFISToken"))
+    }
+
+  override class func setUp() {
+    let options = FirebaseOptions(
+      googleAppID: MockOptions.appID,
+      gcmSenderID: MockOptions.gcmSenderID
+    )
+    options.apiKey = MockOptions.apiKey
+    options.projectID = MockOptions.projectID
+    FirebaseApp.configure(options: options)
+  }
+
+  /// Get model info if server returns a new model info.
+  func testGetModelInfoWith200() {
+    fakeSession.data = fakeRemoteModelInfo.data(using: .utf8)
+    fakeSession.response = HTTPURLResponse(
+      url: URL(string: "www.fake-download-url.com")!,
+      statusCode: 200,
+      httpVersion: nil,
+      headerFields: ["Etag": "fakeModelHash"]
+    )
+    fakeSession.error = nil
+
+    let modelInfoRetriever = ModelInfoRetriever(
+      modelName: "fakeModelName",
+      projectID: "fakeProjectID",
+      apiKey: "fakeAPIKey",
+      authTokenProvider: successAuthTokenProvider,
+      appName: "fakeAppName",
+      session: fakeSession
+    )
+
+    modelInfoRetriever.downloadModelInfo { result in
+      switch result {
+      case let .success(fakemodelInfoResult):
+        switch fakemodelInfoResult {
+        case let .modelInfo(remoteModelInfo):
+          XCTAssertEqual(remoteModelInfo.name, self.fakeModelName)
+          XCTAssertEqual(remoteModelInfo.downloadURL.absoluteString, self.fakeDownloadURL)
+          XCTAssertEqual(remoteModelInfo.size, self.fakeModelSize)
+          XCTAssertEqual(remoteModelInfo.modelHash, self.fakeModelHash)
+        case .notModified: XCTFail("Expected new model info from server.")
+        }
+      case let .failure(error): XCTFail("Failed with error - \(error)")
+      }
+    }
+  }
+
+  /// Get model info if model info is not modified.
+  func testGetModelInfoWith304() {
+    fakeSession.data = fakeRemoteModelInfo.data(using: .utf8)
+    fakeSession.response = HTTPURLResponse(
+      url: URL(string: "www.fake-download-url.com")!,
+      statusCode: 304,
+      httpVersion: nil,
+      headerFields: ["Etag": "fakeModelHash"]
+    )
+    fakeSession.error = nil
+
+    let fakeLocalModelInfo = LocalModelInfo(
+      name: "fakeModelName",
+      modelHash: "fakeModelHash",
+      size: 20,
+      path: "fakeModelPath"
+    )
+
+    let modelInfoRetriever = ModelInfoRetriever(
+      modelName: "fakeModelName",
+      projectID: "fakeProjectID",
+      apiKey: "fakeAPIKey",
+      authTokenProvider: successAuthTokenProvider,
+      appName: "fakeAppName",
+      localModelInfo: fakeLocalModelInfo,
+      session: fakeSession
+    )
+
+    modelInfoRetriever.downloadModelInfo { result in
+      switch result {
```

Comment:
If the completion is not called the test will still succeed. We need to add an expectation like in line 330 check if the completion was called.

Reply:
Good catch, done.